src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -11,13 +11,64 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
+    /// <summary>
+    /// A builder for lazy instances of TypeSymbolWithAnnotations.
+    /// </summary>
+    [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
+    internal struct TypeSymbolWithAnnotationsBuilder
+    {
+        private TypeSymbol _defaultType;
+        private INonNullTypesContext _nonNullTypesContext;
+        private bool _isAnnotated;
+        private TypeSymbolWithAnnotationsImplementation _implementation;
+
+        internal TypeSymbol DefaultType => _defaultType;
+        internal INonNullTypesContext NonNullTypesContext => _nonNullTypesContext;
+        internal bool IsAnnotated => _isAnnotated;
+        internal TypeSymbolWithAnnotationsImplementation Implementation => _implementation;
+
+        internal bool IsNull => _defaultType is null;
+
+        internal bool InterlockedInitialize(TypeSymbolWithAnnotations type)
+        {
+            if ((object)_defaultType != null)
+            {
+                return false;
+            }
+            _isAnnotated = type.IsAnnotated;
+            Interlocked.CompareExchange(ref _nonNullTypesContext, type.NonNullTypesContext, null);
+            Interlocked.CompareExchange(ref _implementation, type.Implementation, null);
+            return (object)Interlocked.CompareExchange(ref _defaultType, type.DefaultType, null) == null;
```

Comment:
>return [](start = 12, length = 6)

Does anyone care for the return value?

Reply:
Yes, several callers decide whether to perform additional checks based on the return value.

---
In reply to: [206366336](https://github.com/dotnet/roslyn/pull/28943#discussion_r206366336) [](ancestors = 206366336)